### PR TITLE
[AKS] support csi drivers extensibility

### PR DESCRIPTION
**Pull request by ZeroMagic**
**Related command**

az aks create [--disable-disk-driver] [--disable-file-driver] [--disable-snapshot-controller]
az aks update [--disable-disk-driver] [--disable-file-driver] [--disable-snapshot-controller] [--enable-disk-driver] [--enable-file-driver] [--enable-snapshot-controller]

**Description**

With the feature, customers can enable/disable the CSI Drivers in their clusters according to demand.
**Testing Guide**

pyttest test_managed_cluster_decorator.py

azdev test acs.test_aks_create_and_update_with_csi_drivers_extensibility --live 
azdev test acs.test_aks_create_with_standard_csi_drivers --live 

**History Notes**

[AKS] `az aks create/update`: Add new parameters `--enable-disk-driver` and `--disable-disk-driver` to enable/disable AzureDisk CSI Driver. When creating new cluster, AzureDisk CSI Driver is enabled by default.
[AKS] `az aks create/update`: Add new parameters `--enable-file-driver` and `--disable-file-driver` to enable/disable AzureFile CSI Driver. When creating new cluster, AzureFile CSI Driver is enabled by default.
[AKS] `az aks create/update`: Add new parameters `--enable-snapshot-controller` and `--disable-snapshot-controller` to enable/disable CSI Snapshot Controller. When creating new cluster, CSI Snapshot Controller is enabled by default.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/_params.py [hunk 1]
```diff
@@ -290,6 +290,12 @@ def load_arguments(self, _):
         c.argument('node_resource_group')
         c.argument('enable_defender', action='store_true')
         c.argument('defender_config', validator=validate_defender_config_parameter)
+        c.argument('enable_disk_driver', action='store_true')
+        c.argument('disable_disk_driver', action='store_true')
+        c.argument('enable_file_driver', action='store_true')
+        c.argument('disable_file_driver', action='store_true')
+        c.argument('enable_snapshot_controller', action='store_true')
+        c.argument('disable_snapshot_controller', action='store_true')
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_params.py:298]: This is under the scope of `aks create`, should declare disable_xxx only and missing enable_xx & disable_xxx for `aks update`

**Review by cvvz — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py [hunk 2]
```diff
@@ -4388,6 +4518,17 @@ def set_up_windows_profile(self, mc: ManagedCluster) -> ManagedCluster:
...
         return mc
 
+    def set_up_storage_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        """Set up storage profile for the ManagedCluster object.
+
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        mc.storage_profile = self.context.get_storage_profile()
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py:4528]: [CI](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1804092&view=logs&j=4e4d232d-230a-55c9-0730-4c1a9c36b9e5&t=6d1bdb99-245b-5117-da18-01d8df6f72c0&l=1967) failed due to no such profile in API version `2020-11-01`. A simple workaround would be add a `hasattr` check here.

```suggestion
        if hasattr(self.models, "ManagedClusterStorageProfile"):
            mc.storage_profile = self.context.get_storage_profile()
```

File: src/azure-cli/azure/cli/command_modules/acs/_params.py (hunk 1 above)
Comment (ZeroMagic) [reply src/azure-cli/azure/cli/command_modules/acs/_params.py]: done

File: src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py (hunk 2 above)
Comment (ZeroMagic) [reply src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py]: done.

**Review by FumingZhang — APPROVED** [review]
LGTM

**Comment by ZeroMagic** [comment]
Could you have a look @zhoxing-ms

File: src/azure-cli/azure/cli/command_modules/acs/_params.py [hunk 3]
```diff
@@ -290,6 +290,9 @@ def load_arguments(self, _):
         c.argument('node_resource_group')
         c.argument('enable_defender', action='store_true')
         c.argument('defender_config', validator=validate_defender_config_parameter)
+        c.argument('disable_disk_driver', action='store_true')
+        c.argument('disable_file_driver', action='store_true')
+        c.argument('disable_snapshot_controller', action='store_true')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/_params.py:295]: Just out of curiosity, may I ask why is the design of this set of parameters not consistent with the following combination to support the `enable_xxx` parameters?

**Comment by zhoxing-ms** [comment]
Could you write the history notes for the feature of these new parameters in the PR description like #23324
For more details, please refer to this doc [submitting-pull-requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests)

File: src/azure-cli/azure/cli/command_modules/acs/_params.py (hunk 3 above)
Comment (ZeroMagic) [reply src/azure-cli/azure/cli/command_modules/acs/_params.py:295]: These drivers are enabled by default. So when creating, we just need to provide disable flag. In addition, here is the preview PR https://github.com/Azure/azure-cli-extensions/pull/4742. It includes more context. We have discussed before.

**Comment by ZeroMagic** [comment]
> Could you write the history notes for the feature of these new parameters in the PR description like #23324 For more details, please refer to this doc [submitting-pull-requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests)

Updated the description

File: src/azure-cli/azure/cli/command_modules/acs/_params.py (hunk 3 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/acs/_params.py:295]: Got it, thanks~

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**